pkg/splunk/test/awss3client.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2018-2021 Splunk Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+	enterprisev1 "github.com/splunk/splunk-operator/pkg/apis/enterprise/v1"
+)
+
+// MockAWSS3Object struct contains contents returned as part of S3 response
+type MockAWSS3Object struct {
+	Etag         *string
+	Key          *string
+	LastModified *time.Time
+	Size         *int64
+	StorageClass *string
+}
+
+// MockAWSS3Client is used to store all the objects for an app source
+type MockAWSS3Client struct {
+	Objects []*MockAWSS3Object
+}
+
+// MockAWSS3Handler is used for checking response received
+type MockAWSS3Handler struct {
+	WantSourceAppListResponseMap map[string]MockAWSS3Client
+	GotSourceAppListResponseMap  map[string]MockAWSS3Client
+}
+
+// AddObjects adds mock AWS S3 Objects to handler
+func (c *MockAWSS3Handler) AddObjects(appFrameworkRef enterprisev1.AppFrameworkSpec, objects ...MockAWSS3Client) {
+	for n := range objects {
+		mockAWSS3Client := objects[n]
+		appSource := appFrameworkRef.AppSources[n]
+		if c.WantSourceAppListResponseMap == nil {
+			c.WantSourceAppListResponseMap = make(map[string]MockAWSS3Client)
+		}
+		c.WantSourceAppListResponseMap[appSource.Name] = mockAWSS3Client
+	}
+}
+
+// CheckAwsS3Response checks if the received objects are same as the one we expect
+func (c *MockAWSS3Handler) CheckAwsS3Response(t *testing.T, testMethod string) {
```

Comment:
Probably should make the `AWS` to be consistent with the other places in the S3Client code